### PR TITLE
Updated docstrings

### PR DESCRIPTION
**Pull request by ubyndr**
Draft

### CODE REVIEW

File: src/pandasaurus/query.py [hunk 1]
```diff
@@ -11,19 +11,37 @@
 
 
 class Query:
-    """Query class is responsible for returning the non-redundant graph for s subClassOf o as a simple Pandas dataframe
-    with given 2 seeds of classes, S(s) and S(o) from an initial seed, S(i)
+    """Query class is responsible for generating a Pandas dataframe that enriches the seed list with synonyms and
```
Comment (bvarner-ebi) [inline src/pandasaurus/query.py:14]:
```suggestion
    """Query class is responsible for generating a pandas DataFrame that enriches the seed list with synonyms and
```

File: src/pandasaurus/query.py [hunk 2]
```diff
@@ -11,19 +11,37 @@
...
+    all inferred subClassOf relationships between terms in the seed. It also allows queries over the dataframe.
```
Comment (bvarner-ebi) [inline src/pandasaurus/query.py:15]:
```suggestion
    all inferred subClassOf relationships between terms in the seed. It also allows queries over the DataFrame.
```

File: src/pandasaurus/query.py [hunk 3]
```diff
@@ -11,19 +11,37 @@
...
+
+    Attributes:
+        enriched_df (str): Dataframe that is enriched with synonyms and inferred relationships between terms in
```
Comment (bvarner-ebi) [inline src/pandasaurus/query.py:18]:
```suggestion
        enriched_df (str): DataFrame that is enriched with synonyms and inferred relationships between terms in
```

File: src/pandasaurus/query.py [hunk 4]
```diff
@@ -11,19 +11,37 @@
...
+        the seed. It will be used in further filtered queries.
+
     """
 
-    def __init__(self, seed_list: List[str]):
+    def __init__(self, seed_list: List[str], enrichment_property_list: List[str] = []):
+        """A Query object is initialised by passing a list of seed terms (where each term is a CURIE string,
+        e.g. CL:0000001; all OBO standard curies are recognised). It generates a Pandas dataframe that enriches the
+        seed list with synonyms and all inferred subClassOf relationships; by default, and it supports other
```
Comment (bvarner-ebi) [inline src/pandasaurus/query.py:26]:
```suggestion
        seed list with synonyms and all inferred subClassOf relationships by default, and it supports other
```

File: src/pandasaurus/query.py [hunk 5]
```diff
@@ -11,19 +11,37 @@
...
+        relationships; between terms in the seed. Additional methods allow enrichment with terms outside the seed
+        from slims or specified by a semantic context.
+
+        Args:
+            seed_list (str): A list of seed terms where each term is a CURIE string
+            enrichment_property_list (List[str]): Property list to extend enrichment queries.
```
Comment (bvarner-ebi) [inline src/pandasaurus/query.py:32]:
```suggestion
            enrichment_property_list (List[str]): Property list to extend enrichment queries
```

File: src/pandasaurus/query.py [hunk 6]
```diff
@@ -11,19 +11,37 @@
...
+
+        """
         self.seed_list = seed_list
+        self.enrichment_property_list = enrichment_property_list
         self.enriched_df: pd.DataFrame = pd.DataFrame()
 
     def simple_enrichment(self) -> pd.DataFrame:
-        """Returns simple enrichment; S(s) = S(i); S(o) = S(i)
+        """Returns a Dataframe that is enriched with synonyms and inferred relationships between terms in the seed.
```
Comment (bvarner-ebi) [inline src/pandasaurus/query.py:40]:
```suggestion
        """Returns a DataFrame that is enriched with synonyms and inferred relationships between terms in the seed.
```

File: src/pandasaurus/query.py [hunk 7]
```diff
@@ -11,19 +11,37 @@
...
+        Subject and object terms are members of the seed terms.
 
         Returns:
-             pd.Dataframe: Enriched df
+             pd.Dataframe: Enriched Dataframe
```
Comment (bvarner-ebi) [inline src/pandasaurus/query.py:44]:
```suggestion
             pd.Dataframe: Enriched DataFrame
```

File: src/pandasaurus/query.py [hunk 8]
```diff
@@ -33,15 +51,15 @@ def simple_enrichment(self) -> pd.DataFrame:
         return self.enriched_df
 
     def minimal_slim_enrichment(self, slim_list: List[str]) -> pd.DataFrame:
-        """Returns minimal enrichment; S(s) = S(i); S(o) = S(i) + all classes in some specified slims, where class
-        in slim = class tagged with some specified ‘subset’ axiom
-
+        """Returns a Dataframe that is enriched with synonyms and inferred relationships between terms in the seed list
```
Comment (bvarner-ebi) [inline src/pandasaurus/query.py:54]:
```suggestion
        """Returns a DataFrame that is enriched with synonyms and inferred relationships between terms in the seed list
```

File: src/pandasaurus/query.py [hunk 9]
```diff
@@ -51,15 +69,16 @@ def minimal_slim_enrichment(self, slim_list: List[str]) -> pd.DataFrame:
         return self.enriched_df
 
     def full_slim_enrichment(self, slim_list: List[str]) -> pd.DataFrame:
-        """Returns full slim enrichment; S(s) = S(i); S(o) = S(i) + all classes in some specified slims, where class
-         in slim = class tagged with some specified ‘subset’ axiom, with transitive query of redundant graph such as
-         owl:subClassOf*
+        """Returns a Dataframe that is enriched with synonyms and inferred relationships between terms in the seed list
```
Comment (bvarner-ebi) [inline src/pandasaurus/query.py:72]:
```suggestion
        """Returns a DataFrame that is enriched with synonyms and inferred relationships between terms in the seed list
```

File: src/pandasaurus/query.py [hunk 10]
```diff
@@ -51,15 +69,16 @@ def minimal_slim_enrichment(self, slim_list: List[str]) -> pd.DataFrame:
...
+        given slim lists, classes tagged with some specified ‘subset’ axiom, with inferred terms via transitive
+        subClassOf queries.
 
         Args:
-             slim_list (List[str]): Slim list
+             slim_list (List[str]): Slim list that consists of classes tagged with some specified ‘subset’ axiom
 
          Returns:
-             pd.Dataframe: Enriched df
+             pd.Dataframe: Enriched Dataframe
```
Comment (bvarner-ebi) [inline src/pandasaurus/query.py:81]:
```suggestion
             pd.Dataframe: Enriched DataFrame
```

File: src/pandasaurus/query.py [hunk 11]
```diff
@@ -69,15 +88,16 @@ def full_slim_enrichment(self, slim_list: List[str]) -> pd.DataFrame:
         return self.enriched_df
 
     def contextual_slim_enrichment(self, context: List[str]) -> pd.DataFrame:
-        """Returns contextual slim enrichment; S(s) = S(i); S(o) = S(i) + all classes satisfied by some set of
-        existential restrictions in the ubergraph redundant graph (e.g. part_of 'Kidney')
+        """Returns a Dataframe that is enriched with synonyms and inferred relationships between terms in the seed list
```
Comment (bvarner-ebi) [inline src/pandasaurus/query.py:91]:
```suggestion
        """Returns a DataFrame that is enriched with synonyms and inferred relationships between terms in the seed list
```

File: src/pandasaurus/query.py [hunk 12]
```diff
@@ -69,15 +88,16 @@ def full_slim_enrichment(self, slim_list: List[str]) -> pd.DataFrame:
...
+        and in an extended seed list. The extended seed list consists of terms from the seed list and all terms
+        satisfied by some set of existential restrictions in the ubergraph (e.g. part_of 'Kidney').
```
Comment (bvarner-ebi) [inline src/pandasaurus/query.py:93]:
```suggestion
        satisfied by some set of existential restrictions in the ubergraph (e.g. part_of 'kidney').
```

File: src/pandasaurus/query.py [hunk 13]
```diff
@@ -69,15 +88,16 @@ def full_slim_enrichment(self, slim_list: List[str]) -> pd.DataFrame:
...
 
         Args:
             context: Organ/tissue/multicellular anatomical structure list to determine the redundant graph via
             existential restrictions
 
         Returns:
-            pd.Dataframe: Enriched df
+            pd.Dataframe: Enriched Dataframe
```
Comment (bvarner-ebi) [inline src/pandasaurus/query.py:100]:
```suggestion
            pd.Dataframe: Enriched DataFrame
```

**Review by bvarner-ebi — APPROVED** [review]
Please see inline suggestions.

File: src/pandasaurus/query.py [hunk 14]
```diff
@@ -33,15 +51,15 @@ def simple_enrichment(self) -> pd.DataFrame:
...
-        """Returns minimal enrichment; S(s) = S(i); S(o) = S(i) + all classes in some specified slims, where class
-        in slim = class tagged with some specified ‘subset’ axiom
-
+        """Returns a DataFrame that is enriched with synonyms and inferred relationships between terms in the seed list
+        and in an extended seed list. The extended seed list consists of terms from the seed list and terms from
+        given slim lists, classes tagged with some specified ‘subset’ axiom.
 
         Args:
-            slim_list (List[str]): Slim list
+            slim_list: Slim list that consists of classes tagged with some specified ‘subset’ axiom
```
Comment (dosumis) [inline src/pandasaurus/query.py:59]: slim list = list 'subset' tags.  this makes is sound like a list of IDs

File: src/pandasaurus/query.py [hunk 15]
```diff
@@ -11,19 +11,37 @@
...
+    def __init__(self, seed_list: List[str], enrichment_property_list: Optional[List[str]] = None):
+        """A Query object is initialised by passing a list of seed terms (where each term is a CURIE string,
+        e.g. CL:0000001; all OBO standard CURIESs are recognised). It generates a pandas DataFrame that enriches the
+        seed list with synonyms and all inferred subClassOf relationships by default, and it supports other
+        relationships between terms in the seed. Additional methods allow enrichment with terms outside the seed
+        from slims or specified by a semantic context.
+
+        Args:
+            seed_list: A list of seed terms where each term is a CURIE string
+            enrichment_property_list: Property list to extend enrichment queries. Defaults to None
```
Comment (dosumis) [inline src/pandasaurus/query.py:32]: In standard Python terminology, the second argument is a KWARG (keyword argument).  That it defaults to `None` is an internal implementation issue and would not normally be mentioned.

**Review by dosumis — CHANGES_REQUESTED** [review]
Some minor tweaks.  Then happy to merge.

**Review by dosumis — APPROVED** [review]

**Pull request merged**